pkg/routeagent/main.go

```diff
@@ -27,10 +27,11 @@ var (
 )
 
 type SubmarinerRouteControllerSpecification struct {
-	ClusterID   string
-	Namespace   string
-	ClusterCidr []string
-	ServiceCidr []string
+	ClusterID             string
+	Namespace             string
+	ClusterCidr           []string
+	ServiceCidr           []string
+	ResyncIntervalMinutes int `default:"0"`
```

Comment:
I prefer we do not expose this, at least until we make the code handle resync properly and see a need for it.

Reply:
I feel safer keeping it there, what if our change has unintended consequences, or if we introduce a bug which could be reverted by tweaking the setting. What if we discover this on a production deployment? (apparently there are production deployments now out there..) 

I think that's may be far now. But I've been able to provide workarounds for customers by settings of this type.

Reply:
For ease of mind I debug-logged what the nodes do during resync (I set 1min manually):

The non-gateway:

```
I0214 12:20:04.031327       1 route.go:512] Enqueueing endpoint for route controller {"cluster_id":"cluster3","cable_name":"submariner-cable-cluster3-172-17-0-4","hostname":"cluster3-worker","subnets":["100.96.0.0/16","10.246.0.0/16"],"private_ip":"172.17.0.4","public_ip
":"","nat_enabled":false,"backend":"ipsec"}
I0214 12:20:04.031524       1 route.go:512] Enqueueing endpoint for route controller {"cluster_id":"cluster2","cable_name":"submariner-cable-cluster2-172-17-0-3","hostname":"cluster2-worker","subnets":["100.95.0.0/16","10.245.0.0/16"],"private_ip":"172.17.0.3","public_ip
":"","nat_enabled":false,"backend":"ipsec"}
I0214 12:20:04.042278       1 route.go:429] Updating iptable rules for remote cluster's Endpoint "cluster3-submariner-cable-cluster3-172-17-0-4" with subnets [100.96.0.0/16 10.246.0.0/16]
I0214 12:20:04.050802       1 route.go:436] Processing local Endpoint cluster2-submariner-cable-cluster2-172-17-0-3 with IP 172.17.0.3, Host cluster2-worker for this cluster
I0214 12:20:04.050898       1 route.go:464] This route agent is running on a non-gateway/non-active node
I0214 12:20:04.050948       1 route.go:475] Creating the vxlan interface vx-submariner with gateway node IP 172.17.0.3
I0214 12:20:04.051597       1 vxlan.go:62] VxLAN interface already exists with same configuration.
I0214 12:20:04.059206       1 route.go:671] Reconciling routes to gw: 240.17.0.3
I0214 12:20:04.059558       1 route.go:687] Processing route {Ifindex: 7680 Dst: 10.246.0.0/16 Src: <nil> Gw: 240.17.0.3 Flags: [] Table: 254}
I0214 12:20:04.059702       1 route.go:692] Found route {Ifindex: 7680 Dst: 10.246.0.0/16 Src: <nil> Gw: 240.17.0.3 Flags: [] Table: 254} with gw 240.17.0.3 already installed
I0214 12:20:04.059726       1 route.go:687] Processing route {Ifindex: 7680 Dst: 100.96.0.0/16 Src: <nil> Gw: 240.17.0.3 Flags: [] Table: 254}
I0214 12:20:04.059772       1 route.go:692] Found route {Ifindex: 7680 Dst: 100.96.0.0/16 Src: <nil> Gw: 240.17.0.3 Flags: [] Table: 254} with gw 240.17.0.3 already installed
I0214 12:20:04.059787       1 route.go:687] Processing route {Ifindex: 7680 Dst: 240.0.0.0/8 Src: 240.17.0.5 Gw: <nil> Flags: [] Table: 254}
I0214 12:20:04.059814       1 route.go:689] Found nil gw or dst
I0214 12:20:04.060107       1 route.go:728] Found equivalent route, not adding
I0214 12:20:04.060299       1 route.go:728] Found equivalent route, not adding
I0214 12:20:04.060355       1 route.go:493] Successfully processed local Endpoint "cluster2-submariner-cable-cluster2-172-17-0-3"
```
On the gw:

```
I0214 12:20:04.032232       1 route.go:512] Enqueueing endpoint for route controller {"cluster_id":"cluster2","cable_name":"submariner-cable-cluster2-172-17-0-3","hostname":"cluster2-worker","subnets":["100.95.0.0/16","10.245.0.0/16"],"private_ip":"172.17.0.3","public_ip
":"","nat_enabled":false,"backend":"ipsec"}
I0214 12:20:04.032376       1 route.go:512] Enqueueing endpoint for route controller {"cluster_id":"cluster3","cable_name":"submariner-cable-cluster3-172-17-0-4","hostname":"cluster3-worker","subnets":["100.96.0.0/16","10.246.0.0/16"],"private_ip":"172.17.0.4","public_ip
":"","nat_enabled":false,"backend":"ipsec"}
I0214 12:20:04.042951       1 route.go:436] Processing local Endpoint cluster2-submariner-cable-cluster2-172-17-0-3 with IP 172.17.0.3, Host cluster2-worker for this cluster
I0214 12:20:04.043036       1 route.go:446] This route agent is running on the active gateway node
I0214 12:20:04.043079       1 route.go:452] Creating the vxlan interface: vx-submariner on the gateway node
I0214 12:20:04.043814       1 vxlan.go:62] VxLAN interface already exists with same configuration.
I0214 12:20:04.044028       1 vxlan.go:158] Successfully added the bridge fdb entry 172.17.0.5 00:00:00:00:00:00
I0214 12:20:04.044191       1 vxlan.go:158] Successfully added the bridge fdb entry 172.17.0.3 00:00:00:00:00:00
I0214 12:20:04.044379       1 route.go:306] Successfully configured rp_filter to loose mode(2) on vx-submariner
I0214 12:20:04.045028       1 route.go:619] Processing route {Ifindex: 7680 Dst: 240.0.0.0/8 Src: 240.17.0.3 Gw: <nil> Flags: [] Table: 254}
I0214 12:20:04.045099       1 route.go:621] Found nil gw or dst
I0214 12:20:04.053169       1 route.go:429] Updating iptable rules for remote cluster's Endpoint "cluster3-submariner-cable-cluster3-172-17-0-4" with subnets [100.96.0.0/16 10.246.0.0/16]
```

There seems to be no destructive event, so I think it could potentially be of help if anything arises (stale connections?, does the watcher ping the http connection being watched?, can be remediated with TCP keepalived settings though..)

Reply:
We're also cleaning routes and XRFM policies. `processNextEndpoint` is really coded assuming `handleRemovedEndpoint` occurred prior. I understand what your saying about making it configurable but I'd be wary of turning this on in a production environment w/o fully understanding its effects and testing it even though it's already been on so it would **seem** to be OK. This also would it to the public API so we'd be stuck with it. I'd still prefer not to add the config at this point but if others agree we should then I'm OK.

Reply:
Actually `SubmarinerRouteControllerSpecification ` is not really public so this would effectively be a hidden config. So it should be fine should we want to remove it at some point.

Reply:
Right, I don't propose advertising or documenting it, and it we do, under the proper warning. I'm testing it though in a loop with 1 min resync, and the gw failover/connectivity does not seem to fail.

I think the "220 table" and your "workqueue" patches actually fixed it for good/now ;)